### ISSUE TITLE
Unsupported regex EXPRESSION

### ISSUE THREAD

**Issue by loicgasser** [geographika/mappyfile#27]
```python
layer = """
LAYER
  CLASSITEM "objval"
  CONNECTION "user=user password=password host=host port=5432 dbname=mydb"
  CONNECTIONTYPE POSTGIS
  DATA "the_geom from vec200_railway using unique gtdboid using srid=21781"
  GROUP "Vector200"
  METADATA
    "wms_srs" "EPSG:21781 EPSG:2056 EPSG:4326 EPSG:3857 EPSG:3034 EPSG:3035 EPSG:4258 EPSG:31287 EPSG:25832 EPSG:25833 EPSG:31467 EPSG:32632 EPSG:32633"
    "wms_group_title" "vec200"
    "wms_title" "Vector200 railway"
    "gml_featureid" "gtdboid"
    "ows_include_items"   "all"
    "ows_exclude_items"   "bgdi_modified,bgdi_created,bgdi_modified_by,bgdi_created_by"
  END
  TEMPLATE "ttt"
  NAME "ch.swisstopo.vec200-transportation-railway"
  STATUS OFF
  TEMPLATE "ttt"
  TYPE LINE
  UNITS METERS
  CLASS
    NAME "NS_Bahn_NS_BahnAuto"
    EXPRESSION /NS_Bahn|NS_BahnAuto/
    STYLE
      ANGLE 360
      COLOR 230 0 0
      OPACITY 100
      WIDTH 3
    END
  END
END
"""
import mappyfile

mappyfile.loads(layer)
```

It fails. See: http://mapserver.org/fr/mapfile/expressions.html#list-expressions

`{NS_Bahn,NS_BahnAuto}` does work but `/NS_Bahn|NS_BahnAuto/` does not.

List expressions seem to be better anyway, but it is still a valid expression.

**Comment by loicgasser** [comment]
Not sure why but
```python
expr = '''
  CLASS
    EXPRESSION ('[construct]' ~* /Br.*$/)
    STYLE
      ANGLE 360
      COLOR 0 0 0
      OPACITY 100
      SIZE 6
      SYMBOL "circle"
    END
  END
'''
import mappyfile
mappyfile.loads(expr)
```

while fail and output:
```
lark.lexer.UnexpectedInput: No token defined for: '*' in '*$/)\n' at line 5
```

This expression:

```python
expr = '''
  CLASS
    EXPRESSION ('[construct]' ~* /^Br.*/)
    STYLE
      ANGLE 360
      COLOR 0 0 0
      OPACITY 100
      SIZE 6
      SYMBOL "circle"
    END
  END
'''
import mappyfile
mappyfile.loads(expr)
```

will succeed.
Both are valid regular expressions. I tried to debug but not luck...

I think there is some kind of ambiguity that is not resolved properly.
See also latest comment here:
https://github.com/erezsh/lark/issues/16#issuecomment-320521453

**Comment by erezsh** [comment]
Hi, author of Lark here, this is not that kind of ambiguity.
I believe the reason for this is a collision between two tokens: REGEXP1 and PATH
@geographika If my guess is correct, my suggestion is to catch it and fall back to Earley. (It might be impossible to solve this collision for every situation using a traditional lexer)

**Comment by loicgasser** [comment]
Thanks for the heads up @erezsh 
On master it's using earley per default.
I believe an `EXPRESSION` will never contain `PATH` so there must be a way to express that in the grammar. ping @geographika

**Comment by erezsh** [comment]
@loicgasser Sorry, you're right, mappyfile uses Earley with a traditional lexer as well. I'll give it some more thought.

**Comment by geographika** [comment]
@loicgasser - I've added a new test for this - test_expression(), see https://github.com/geographika/mappyfile/blob/master/tests/test_snippets.py#L669

I've also updated master to use the lalr parser, with fallbacks to Earley. Does this work for you?

**Comment by loicgasser** [comment]
Cool, I ll give it a try tomorrow. Thank you so much for that.

**Comment by geographika** [comment]
@loicgasser - ignore my last comment - I was testing the wrong expression. I'll look at the list one again today.

**Comment by geographika** [comment]
Ok I can recreate this and it fails with both lalr and Earley parsing. I added a test at https://github.com/geographika/mappyfile/blob/master/tests/test_snippets.py#L680

**Comment by loicgasser** [comment]
Ok, this one is probably a little tricky. I am gona try to help and narrow it down. I am not super familiar with all the grammar syntax, but as I said before, if we can constrain what should follow an expression I think it could help.

**Comment by erezsh** [comment]
Solved in branch lark_0.3.6

**Comment by geographika** [comment]
Thanks @erezsh! Merged and added to a new v04.2 release. 
@loicgasser the examples you gave at the top now parse without problems at http://mappyfile.geographika.net/

**Comment by loicgasser** [comment]
Great thanks! Tested and it works as expected.
